server/src/main/java/org/opensearch/action/search/SearchRequestSlowLog.java

```diff
@@ -233,18 +235,22 @@ private static String escapeJson(String text) {
 
     void setWarnThreshold(TimeValue warnThreshold) {
         this.warnThreshold = warnThreshold.nanos();
+        setEnabled();
```

Comment:
nit: Can name this to `setEnabledIfThresholdExceed()` or something similar to make it more readable?